Addon-docs: Fix React Suspense in source snippets.

zhyd1997 opened the pull request:
Issue: #11554

## What I did

add `<React.Suspense>` support in `source snippets.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

zhyd1997 committed:
Addon-docs: Fix React Suspense in source snippets.

zhyd1997 commented:
Refactored PR: https://github.com/storybookjs/storybook/pull/19785

zhyd1997 committed:
fix: added isSuspense and isProfiler type.

shilman commented on code/renderers/react/src/docs/jsxDecorator.test.tsx:
```diff
@@ -155,6 +155,23 @@ describe('renderJsx', () => {
+    expect(renderJsx(createElement(SuspenseComponent, {}, 'I am Suspense'), {}))
+      .toMatchInlineSnapshot(`
+      <SuspenseComponent>
+        I am Suspense
```
Any ideas why the `<div>` is not showing up?

zhyd1997 replied:
Hi @shilman 

I submitted a refactor PR: #19785 to handle other "know symbols", but have some issues on CI, you can check it.

zhyd1997 replied:
Uhh, `div`, sorry, it's my problem, I will add it when I come back my office.

zhyd1997 replied:
Actually It works for new creat-react-app when I added them in bundled code, I will figure out why the test passed with wrong snapshot.

shilman reviewed (commented):
Thanks for fixing this @zhyd1997 !! Is there a full list of these "known symbols" somewhere & can we do all of them in this PR?

zhyd1997 commented:
Hi @shilman!

with the help of @JReinhold on Discord, I have fixed the CI errors on PR #19785 

- added a new react type support -- `StrictMode`, other types will be added later, and
- in my opinion, the `div` missing is another issue...

so could we talk about the feature/bug on PR #19785 ?

ndelangen committed:
Merge branch 'next' into feat/c0d4f587-yadong-react-supsense

ndelangen commented:
@zhyd1997 Is the other PR a replacement for this one?

Can I close this PR, then?

zhyd1997 commented:
Hi @ndelangen 

The replacement: https://github.com/storybookjs/storybook/pull/19785
